pyccel/codegen/printing/ccode.py

```diff
@@ -2084,12 +2084,22 @@ def _print_NumpySign(self, expr):
         self.add_import(c_imports['pyc_math_c'])
         primitive_type = expr.dtype.primitive_type
         func = ''
+        arg = self._print(expr.args[0])
```

Comment:
`self._print(expr.args[0])` is also called on line 2104. It would be better to only call this once